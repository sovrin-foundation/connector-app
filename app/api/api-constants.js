// @flow

export const TOKEN_EXPIRED_CODE = 'VES-101'
export const SERVER_ERROR_CODE = 'OCS-000'
export const SERVER_API_CALL_ERROR = 'OCS-001'
export const PENDING_CONNECTION_REQUEST_CODE = 'CRS-103'
export const PUSH_NOTIFICATION_SENT_CODE = 'push-notification-sent'

export const QR_CODE_REQUEST_ID = 'id'
export const QR_CODE_SENDER_DETAIL = 's'
export const QR_CODE_SENDER_DID = 'd'
export const QR_CODE_SENDER_NAME = 'n'
export const QR_CODE_LOGO_URL = 'l'
export const QR_CODE_SENDER_VERIFICATION_KEY = 'v'
export const QR_CODE_SENDER_KEY_DELEGATION = 'dp'
export const QR_CODE_DELEGATION_DID = 'd'
export const QR_CODE_DELEGATION_KEY = 'k'
export const QR_CODE_DELEGATION_SIGNATURE = 's'
export const QR_CODE_SENDER_AGENCY = 'sa'
export const QR_CODE_SENDER_AGENCY_DID = 'd'
export const QR_CODE_SENDER_AGENCY_KEY = 'v'
export const QR_CODE_SENDER_AGENCY_ENDPOINT = 'e'
export const QR_CODE_TARGET_NAME = 't'
export const GENESIS_FILE_NAME = 'pool_transactions_genesis'

export const MESSAGE_TYPE = {
  AUTH: 'auth-req',
  CLAIM_OFFER: 'credOffer',
  CLAIM_REQUEST: 'credReq',
  PROOF_REQUEST: 'proofReq',
  PROOF: 'proof',
  CLAIM: 'cred',
}

export const PAYLOAD_TYPE = {
  GET_MESSAGE: 'GET_MSGS',
  UPDATE_PUSH_COM_METHOD: 'UPDATE_PUSH_COM_METHOD',
  SEND_MESSAGE: 'SEND_MSG',
}

export const STATUS_CODE = {
  ACCEPT: 'MS-104',
}

export const ERROR_ALREADY_EXIST = {
  code: 'OCS',
  message: 'You already have a relationship with ',
}

export const ERROR_ALREADY_EXIST_TITLE = 'Relationship already established'

export const ERROR_INVITATION_RESPONSE_PARSE_CODE = 'RN-100'

export const ERROR_INVITATION_RESPONSE_PARSE =
  'Error parsing error message in invitation response'

export const ERROR_PENDING_INVITATION_RESPONSE_PARSE_CODE = 'RN-101'

export const ERROR_PENDING_INVITATION_RESPONSE_PARSE =
  'Error parsing error message Pending Invitation SMS'

export const ERROR_INVITATION_RESPONSE_FAILED =
  'Failed to establish connection. Please try again later.'
