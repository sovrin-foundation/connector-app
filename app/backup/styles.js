// @flow
import { StyleSheet, Dimensions, Platform } from 'react-native'
import {
  PADDING_HORIZONTAL,
  PADDING_VERTICAL,
  isBiggerThanShortDevice,
  color,
  grey,
} from '../common/styles/constant'

const { width, height } = Dimensions.get('window')
const SPACE_FILLER = 50
const amountSpacing = isBiggerThanShortDevice ? PADDING_VERTICAL * 2 : 0
const submitButtonHeight = isBiggerThanShortDevice ? 63 : 43
const showRecoveryButtonHeight = isBiggerThanShortDevice ? 150 : 100
const inputBoxHeight =
  isBiggerThanShortDevice || Platform.OS === 'ios' ? 137 : 40
export const chatBubbleTextOffset = 40
const styles = StyleSheet.create({
  tabContainer: {
    maxHeight: 60,
  },
  backgroundImage: {
    width: '100%',
    flex: 1,
    position: 'absolute',
  },
  headerCloseIcon: {
    marginRight: PADDING_HORIZONTAL,
    marginBottom: PADDING_VERTICAL,
    alignSelf: 'flex-end',
  },
  headerIcon: {
    marginRight: isBiggerThanShortDevice ? 5 : 0,
    marginTop: isBiggerThanShortDevice ? 15 : 0,
    alignSelf: 'flex-end',
  },
  headerBackIcon: {
    marginLeft: isBiggerThanShortDevice ? 5 : 0,
    marginTop: isBiggerThanShortDevice ? 15 : 5,
    alignSelf: 'flex-start',
  },
  headerSpacer: {
    height: '100%',
    width: isBiggerThanShortDevice ? SPACE_FILLER : SPACE_FILLER * 3 / 4,
  },
  showRecoveryPassphrase: {
    width: '86%',
    height: showRecoveryButtonHeight,
    borderRadius: 5,
    alignSelf: 'center',
    backgroundColor: color.bg.tertiary.color,
    elevation: 7,
    marginBottom: 60,
  },
  genRecovery: {
    backgroundColor: color.bg.eleventh.color,
  },
  genRecoveryHeader: {
    width: '100%',
  },
  genRecoveryPhraseContainer: {
    position: 'absolute',
    top: 20,
    left: chatBubbleTextOffset / 2,
    paddingHorizontal: '7%',
  },
  genRecoveryPhraseLoadingContainer: {
    padding: 40,
    position: 'absolute',
    top: 0,
  },
  genRecoveryPhrase: {
    textAlign: 'center',
    color: color.bg.eleventh.color,
    fontSize: isBiggerThanShortDevice ? 20 : 18,
    lineHeight: isBiggerThanShortDevice ? 30 : 28,
    fontWeight: '900',
  },
  genRecoveryText: {
    marginBottom: isBiggerThanShortDevice ? 10 : '2%',
    alignSelf: 'center',
    fontSize: isBiggerThanShortDevice ? 22 : 20,
    fontWeight: '600',
  },
  genRecoveryMessage: {
    paddingHorizontal: 20,
    alignSelf: 'center',
    fontSize: isBiggerThanShortDevice ? 18 : 16,
    lineHeight: isBiggerThanShortDevice ? 22 : 20,
    fontWeight: '500',
    marginBottom: isBiggerThanShortDevice ? 20 : 5,
  },
  genRecoverySecondMessage: {
    paddingHorizontal: 20,
    alignSelf: 'center',
    fontSize: isBiggerThanShortDevice ? 18 : 16,
    lineHeight: isBiggerThanShortDevice ? 22 : 20,
    fontWeight: 'bold',
  },
  genRecoverySmallMessage: {
    fontSize: 14,
    lineHeight: 17,
    fontWeight: 'bold',
    marginBottom: isBiggerThanShortDevice ? 15 : '3%',
  },
  verifyMainContainer: {
    flex: 1,
    backgroundColor: color.bg.twelfth.color,
  },
  verifyMainText: {
    paddingHorizontal: 20,
    fontSize: 18,
    lineHeight: 22,
    fontWeight: '500',
    marginTop: isBiggerThanShortDevice ? 40 : 20,
    marginBottom: isBiggerThanShortDevice ? 40 : 20,
  },
  exportBackupMainText: {
    paddingHorizontal: 10,
    fontSize: isBiggerThanShortDevice ? 18 : 16,
    lineHeight: isBiggerThanShortDevice ? 22 : 20,
    fontWeight: '500',
  },
  inputBox: {
    marginBottom: 24,
    marginRight: 20,
    marginLeft: 20,
    height: inputBoxHeight,
    backgroundColor: 'rgba(0,0,0,0.33)',
    color: 'white',
    padding: 10,
    textAlignVertical: 'top',
  },
  submitButton: {
    marginLeft: '2%',
    marginRight: '2%',
    marginBottom: 15,
    borderRadius: 5,
    shadowColor: grey,
    shadowOffset: {
      width: 1,
      height: 1,
    },
    shadowRadius: 5,
    shadowOpacity: 0.3,
    elevation: 7,
    height: isBiggerThanShortDevice ? 53 : 43,
  },
  backgroundImageVerify: {
    flex: 1,
    position: 'absolute',
  },
  exportBackup: {
    backgroundColor: color.bg.thirteenth.color,
  },
  wrapper: {
    paddingHorizontal: PADDING_HORIZONTAL,
  },
  title: {
    fontWeight: '600',
    lineHeight: isBiggerThanShortDevice ? 27 : 22,
    fontSize: isBiggerThanShortDevice ? 22 : 20,
    marginBottom: isBiggerThanShortDevice ? 20 : 10,
    width: '100%',
  },
  exportBackupTitle: {
    fontWeight: '600',
    lineHeight: isBiggerThanShortDevice ? 27 : 22,
    fontSize: isBiggerThanShortDevice ? 22 : 20,
    marginBottom: isBiggerThanShortDevice ? 30 : 10,
    marginTop: isBiggerThanShortDevice ? 10 : 0,
    maxWidth: 290,
  },
  exportBackupText: {
    marginBottom: 20,
    backgroundColor: color.bg.thirteenth.color,
    alignSelf: 'center',
    fontSize: 22,
  },
  exportBackupMessage: {
    backgroundColor: color.bg.thirteenth.color,
    alignSelf: 'center',
    fontSize: 18,
  },
  exportBackupFile: {
    fontSize: 15,
    lineHeight: 18,
    fontWeight: '900',
  },
  exportBackupSmallMessage: {
    fontSize: 14,
    maxWidth: 300,
    lineHeight: 17,
    marginBottom: isBiggerThanShortDevice ? 30 : 10,
  },
  backupComplete: {
    backgroundColor: color.bg.fourteenth.color,
  },
  backupCompleteHeader: {
    width: '100%',
  },
  backupCompleteText: {
    fontSize: 18,
    fontWeight: '500',
    lineHeight: 22,
  },
  alignItemsCenter: {
    marginBottom: 6,
    marginLeft: '5%',
    marginRight: '5%',
  },
  verticalSpacing: {
    paddingTop: amountSpacing,
  },
  customButtonColor: {
    fontWeight: '600',
    fontSize: 18,
  },
  lockIconImage: {
    paddingTop: isBiggerThanShortDevice ? 40 : '2%',
  },
  imageIcon: {
    resizeMode: 'stretch',
    width: '100%',
    marginBottom: isBiggerThanShortDevice ? 20 : 0,
  },
  imageIconEncryptFile: {
    resizeMode: 'contain',
    width: isBiggerThanShortDevice ? '100%' : '80%',
    marginBottom: isBiggerThanShortDevice ? 20 : 0,
  },
})

export default styles
