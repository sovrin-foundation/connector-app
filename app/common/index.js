// @flow
export * from './button-constants'
export * from './message-constants'
export * from './route-constants'
export * from './secure-storage-constants'
export function noop() {}
