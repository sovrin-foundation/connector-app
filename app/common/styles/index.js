// @flow
export * from './constant'
export * from './tab-constants'
