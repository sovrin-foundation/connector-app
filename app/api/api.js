// @flow
import { api, options } from './api-utils'
import type {
  GetInvitationLinkApiData,
  MessageApiData,
  DownloadInvitationApiData,
  SendAuthenticationResponseApiData,
  EnvironmentDetailUrlDownloaded,
} from './type-api'
import { PAYLOAD_TYPE, MESSAGE_TYPE, STATUS_CODE } from './api-constants'

export const getInvitationLink = ({
  agencyUrl,
  smsToken,
}: GetInvitationLinkApiData) =>
  api(`${agencyUrl}/agency/url-mapper/${smsToken}`, options('GET'))

export const invitationDetailsRequest = ({ url }: DownloadInvitationApiData) =>
  api(url, options('GET'))

export const getAdditionalData = ({ agencyUrl, dataBody }: MessageApiData) =>
  api(`${agencyUrl}/agency/route`, options('POST', dataBody))

export const sendAuthenticationRequest = ({
  data: { identifier, dataBody },
  config: { agencyUrl },
}: SendAuthenticationResponseApiData) =>
  api(`${agencyUrl}/agent/${identifier}/auth`, options('PUT', dataBody), true)

export const downloadEnvironmentDetails = (
  url: string
): Promise<EnvironmentDetailUrlDownloaded> => api(url, options())
