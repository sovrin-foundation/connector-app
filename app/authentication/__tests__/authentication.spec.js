// @flow
import React from 'react'
import 'react-native'
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux'
import { authenticationRoute } from '../../common/'
import { PUSH_NOTIFICATION_SENT_CODE } from '../../api/api-constants'
import AuthenticationScreen from '../authentication'
import { color } from '../../common/styles'

describe('<AuthenticationScreen />', () => {
  let store = {}

  beforeAll(() => {
    store = {
      getState() {
        return {
          connections: {
            connectionThemes: {
              default: {
                primary: `rgba(${color.actions.button.primary.rgba})`,
                secondary: `rgba(${color.actions.button.secondary.rgba})`,
              },
            },
          },
          route: {
            currentScreen: authenticationRoute,
          },
          authentication: {
            data: {
              offerMsgTitle: 'Hi There',
              offerMsgText:
                'Suncoast Credit Union (sandbox) wants to connect with you',
              statusCode: PUSH_NOTIFICATION_SENT_CODE,
              logoUrl: null,
              remoteConnectionId: 'B4Y9fhpeHdGHBKKtSgAYrB',
            },
          },
          config: {
            showErrorAlerts: false,
          },
          lock: {
            isTouchIdEnabled: false,
          },
          user: {
            avatarName: undefined,
          },
        }
      },
      subscribe() {
        return jest.fn()
      },
      dispatch() {
        return jest.fn()
      },
    }
  })

  it('should match snapshot', () => {
    const component = renderer.create(
      <Provider store={store}>
        <AuthenticationScreen showErrorAlerts />
      </Provider>
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
