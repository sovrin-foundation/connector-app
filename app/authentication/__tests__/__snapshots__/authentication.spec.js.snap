// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AuthenticationScreen /> should match snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      fifth={true}
      style={
        Object {
          "backgroundColor": "#ffffff",
          "flex": 1,
        }
      }
    >
      <View
        accessibilityLabel="authentication-request-text-container"
        accessible={true}
        hCenter={true}
        style={
          Object {
            "flex": 1,
            "justifyContent": "center",
          }
        }
        testID="authentication-request-text-container"
        useNativeDriver={true}
      >
        <View
          accessibilityLabel="authentication-request-text-message-container"
          accessible={true}
          style={
            Object {
              "flex": 1,
              "justifyContent": "flex-end",
            }
          }
          testID="authentication-request-text-message-container"
        >
          <Animatable.View
            accessibilityLabel="authentication-request-text-container-animation"
            accessible={true}
            animation="bounceInDown"
            delay={100}
            duration={1500}
            testID="authentication-request-text-container-animation"
          >
            <View
              accessibilityLabel="authentication-request-text-container-message-title"
              accessible={true}
              center={true}
              style={
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                }
              }
              testID="authentication-request-text-container-message-title"
            >
              <Text
                accessibilityLabel="authentication-request-text-container-title"
                accessible={true}
                allowFontScaling={false}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "fontSize": 20,
                    },
                    Object {
                      "fontFamily": "Lato",
                    },
                    Object {
                      "backgroundColor": "#ffffff",
                      "color": "#777777",
                    },
                    Object {
                      "fontWeight": "800",
                    },
                    Object {
                      "textAlign": "center",
                    },
                    null,
                    null,
                    null,
                    null,
                    null,
                    null,
                    null,
                    null,
                    null,
                    Object {
                      "lineHeight": 30,
                      "marginHorizontal": 30,
                      "marginVertical": 10,
                    },
                  ]
                }
                testID="authentication-request-text-container-title"
              >
                Hi There
              </Text>
              <Text
                accessibilityLabel="authentication-request-text-container-message"
                accessible={true}
                allowFontScaling={false}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "fontSize": 17,
                    },
                    Object {
                      "fontFamily": "Lato",
                    },
                    Object {
                      "backgroundColor": "#ffffff",
                      "color": "#777777",
                    },
                    Object {
                      "fontWeight": "bold",
                    },
                    Object {
                      "textAlign": "center",
                    },
                    null,
                    null,
                    null,
                    null,
                    null,
                    null,
                    null,
                    null,
                    null,
                    Object {
                      "margin": 10,
                    },
                  ]
                }
                testID="authentication-request-text-container-message"
              >
                Suncoast Credit Union (sandbox) wants to connect with you
              </Text>
            </View>
          </Animatable.View>
        </View>
        <View
          accessibilityLabel="authentication-request-text-container-avatars"
          accessible={true}
          style={
            Object {
              "flex": 1,
            }
          }
          testID="authentication-request-text-container-avatars"
        >
          <Animatable.View
            accessibilityLabel="invitation-text-container-avatars-animation"
            accessible={true}
            animation="bounceInUp"
            delay={300}
            duration={1500}
            testID="invitation-text-container-avatars-animation"
          >
            <View
              center={true}
              spaceBetween={true}
              style={
                Object {
                  "alignItems": "center",
                  "justifyContent": "space-between",
                }
              }
            >
              <View
                accessibilityLabel="invitation-text-avatars-container"
                accessible={true}
                row={true}
                spaceBetween={true}
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                    "marginVertical": 40,
                  }
                }
                testID="invitation-text-avatars-container"
                vCenter={true}
              >
                <View
                  accessibilityComponentType={undefined}
                  accessibilityLabel={undefined}
                  accessibilityTraits={undefined}
                  accessible={true}
                  hitSlop={undefined}
                  nativeID={undefined}
                  onLayout={undefined}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  pointerEvents="box-only"
                  style={undefined}
                  testID={undefined}
                >
                  <View
                    accessibilityLabel="invitation-text-avatars-invitee"
                    accessible={true}
                    imageShadow="imageShadow"
                    style={
                      Object {
                        "borderRadius": 38,
                        "elevation": 3,
                        "shadowColor": "rgba(0, 0, 0, 0.25)",
                        "shadowOffset": Object {
                          "height": 3,
                          "width": 0,
                        },
                        "shadowOpacity": 1,
                        "shadowRadius": 8,
                      }
                    }
                    testID="invitation-text-avatars-invitee"
                  >
                    <Image
                      accessibilityComponentType={undefined}
                      accessibilityLabel={undefined}
                      accessibilityTraits={undefined}
                      accessible={true}
                      hitSlop={undefined}
                      nativeID={undefined}
                      onError={undefined}
                      onLayout={undefined}
                      onLoad={undefined}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      source={1}
                      style={
                        Array [
                          Object {
                            "height": 76,
                            "width": 76,
                          },
                          Object {
                            "borderRadius": 38,
                          },
                        ]
                      }
                      testID={undefined}
                    />
                  </View>
                </View>
                <Image
                  source={1}
                  style={
                    Object {
                      "height": 8,
                      "marginHorizontal": 5,
                      "width": 60,
                    }
                  }
                />
                <View
                  accessibilityLabel="invitation-text-avatars-inviter"
                  accessible={true}
                  imageShadow="imageShadow"
                  style={
                    Object {
                      "borderRadius": 38,
                      "elevation": 3,
                      "shadowColor": "rgba(0, 0, 0, 0.25)",
                      "shadowOffset": Object {
                        "height": 3,
                        "width": 0,
                      },
                      "shadowOpacity": 1,
                      "shadowRadius": 8,
                    }
                  }
                  testID="invitation-text-avatars-inviter"
                >
                  <Image
                    accessibilityComponentType={undefined}
                    accessibilityLabel={undefined}
                    accessibilityTraits={undefined}
                    accessible={true}
                    hitSlop={undefined}
                    nativeID={undefined}
                    onError={undefined}
                    onLayout={undefined}
                    onLoad={undefined}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    source={1}
                    style={
                      Array [
                        Object {
                          "height": 76,
                          "width": 76,
                        },
                        Object {
                          "borderRadius": 38,
                        },
                      ]
                    }
                    testID={undefined}
                  />
                </View>
              </View>
            </View>
          </Animatable.View>
        </View>
      </View>
    </View>
    <RCTSafeAreaView
      style={
        Object {
          "backgroundColor": "rgba(119, 119, 119, 1.0)",
        }
      }
    >
      <View
        fifth={true}
        style={
          Object {
            "backgroundColor": "#ffffff",
          }
        }
      >
        <View
          style={Object {}}
        >
          <View
            customColor={
              Object {
                "backgroundColor": "rgba(119, 119, 119, 1.0)",
              }
            }
            row={true}
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <View
                style={Object {}}
              >
                <View
                  style={
                    Array [
                      Object {
                        "marginLeft": 15,
                        "marginRight": 15,
                      },
                      undefined,
                      Object {
                        "marginLeft": 0,
                        "marginRight": 0,
                      },
                      undefined,
                    ]
                  }
                >
                  <View
                    accessibilityComponentType={undefined}
                    accessibilityLabel="authentication-request-deny"
                    accessibilityTraits={undefined}
                    accessible={true}
                    hasTVPreferredFocus={undefined}
                    hitSlop={undefined}
                    isTVSelectable={true}
                    nativeID={undefined}
                    onLayout={undefined}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Array [
                        Object {
                          "backgroundColor": "transparent",
                        },
                        undefined,
                      ]
                    }
                    testID="authentication-request-deny"
                    tvParallaxProperties={undefined}
                  >
                    <View
                      pointerEvents="box-only"
                      style={
                        Array [
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "#9E9E9E",
                            "flexDirection": "row",
                            "justifyContent": "center",
                            "padding": 19,
                          },
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          Object {
                            "backgroundColor": "rgba(119, 119, 119, 1.0)",
                          },
                          undefined,
                          Object {
                            "padding": 12,
                          },
                          undefined,
                          undefined,
                          undefined,
                          Array [
                            Object {
                              "paddingVertical": 17,
                            },
                          ],
                          false,
                          false,
                        ]
                      }
                    >
                      <Text
                        accessible={true}
                        allowFontScaling={true}
                        ellipsizeMode="tail"
                        style={
                          Array [
                            Object {},
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            Array [
                              Object {
                                "color": "white",
                                "fontSize": 16,
                              },
                              Object {
                                "color": "#ffffff",
                              },
                              Object {
                                "fontSize": 14,
                              },
                              undefined,
                              undefined,
                              undefined,
                              undefined,
                              false,
                            ],
                          ]
                        }
                      >
                        Deny
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
            </View>
            <View
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <View
                style={Object {}}
              >
                <View
                  style={
                    Array [
                      Object {
                        "marginLeft": 15,
                        "marginRight": 15,
                      },
                      undefined,
                      Object {
                        "marginLeft": 0,
                        "marginRight": 0,
                      },
                      undefined,
                    ]
                  }
                >
                  <View
                    accessibilityComponentType={undefined}
                    accessibilityLabel="authentication-request-accept"
                    accessibilityTraits={undefined}
                    accessible={true}
                    hasTVPreferredFocus={undefined}
                    hitSlop={undefined}
                    isTVSelectable={true}
                    nativeID={undefined}
                    onLayout={undefined}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Array [
                        Object {
                          "backgroundColor": "transparent",
                        },
                        Array [
                          Object {
                            "borderLeftColor": "#ffffff",
                            "borderLeftWidth": 0.4,
                          },
                        ],
                      ]
                    }
                    testID="authentication-request-accept"
                    tvParallaxProperties={undefined}
                  >
                    <View
                      pointerEvents="box-only"
                      style={
                        Array [
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "#9E9E9E",
                            "flexDirection": "row",
                            "justifyContent": "center",
                            "padding": 19,
                          },
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          Object {
                            "backgroundColor": "rgba(119, 119, 119, 1.0)",
                          },
                          undefined,
                          Object {
                            "padding": 12,
                          },
                          undefined,
                          undefined,
                          undefined,
                          Array [
                            Object {
                              "paddingVertical": 17,
                            },
                            Object {
                              "borderLeftColor": "#ffffff",
                              "borderLeftWidth": 0.4,
                            },
                          ],
                          false,
                          false,
                        ]
                      }
                    >
                      <Text
                        accessible={true}
                        allowFontScaling={true}
                        ellipsizeMode="tail"
                        style={
                          Array [
                            Object {},
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            Array [
                              Object {
                                "color": "white",
                                "fontSize": 16,
                              },
                              Object {
                                "color": "#ffffff",
                              },
                              Object {
                                "fontSize": 14,
                              },
                              undefined,
                              undefined,
                              Object {
                                "fontWeight": "bold",
                              },
                              undefined,
                              false,
                            ],
                          ]
                        }
                      >
                        Connect
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTSafeAreaView>
  </View>
  <Modal
    animationIn="zoomIn"
    animationOut="zoomOut"
    animationOutTiming={100}
    backdropColor="#ffffff"
    backdropOpacity={1}
    hideModalContentWhileAnimating={true}
    isVisible={false}
    onModalHide={[Function]}
    useNativeDriver={true}
  >
    <View
      fifth={true}
      shadow={true}
      style={
        Object {
          "backgroundColor": "#ffffff",
          "elevation": 10,
          "marginHorizontal": 10,
          "shadowColor": "rgba(0, 0, 0, 0.2)",
          "shadowOffset": Object {
            "height": 10,
            "width": 0,
          },
          "shadowOpacity": 1,
          "shadowRadius": 30,
        }
      }
    >
      <View
        style={
          Object {
            "borderBottomColor": "#3f3f3f",
            "borderBottomWidth": 0.2,
            "paddingVertical": 10,
          }
        }
      >
        <View
          accessibilityLabel="invitation-text-avatars-container"
          accessible={true}
          row={true}
          spaceBetween={true}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginHorizontal": 10,
              "marginVertical": 10,
            }
          }
          testID="invitation-text-avatars-container"
          vCenter={true}
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hitSlop={undefined}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            pointerEvents="box-only"
            style={undefined}
            testID={undefined}
          >
            <View
              accessibilityLabel="avatars-pair-avatars-invitee"
              accessible={true}
              imageShadow={null}
              style={
                Object {
                  "borderRadius": 38,
                }
              }
              testID="avatars-pair-avatars-invitee"
            >
              <Image
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                hitSlop={undefined}
                nativeID={undefined}
                onError={undefined}
                onLayout={undefined}
                onLoad={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                source={1}
                style={
                  Array [
                    Object {
                      "height": 76,
                      "width": 76,
                    },
                    Object {
                      "borderRadius": 38,
                    },
                  ]
                }
                testID={undefined}
              />
            </View>
          </View>
          <Image
            accessibilityLabel="invitation-check-mark"
            accessible={true}
            source={1}
            style={
              Array [
                Object {
                  "height": 22,
                  "width": 30,
                },
                undefined,
              ]
            }
            testID="invitation-check-mark"
          />
          <View
            accessibilityLabel="invitation-text-avatars-inviter"
            accessible={true}
            imageShadow={null}
            style={
              Object {
                "borderRadius": 38,
              }
            }
            testID="invitation-text-avatars-inviter"
          >
            <Image
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              hitSlop={undefined}
              nativeID={undefined}
              onError={undefined}
              onLayout={undefined}
              onLoad={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              source={1}
              style={
                Array [
                  Object {
                    "height": 76,
                    "width": 76,
                  },
                  Object {
                    "borderRadius": 38,
                  },
                ]
              }
              testID={undefined}
            />
          </View>
        </View>
        <Text
          accessibilityLabel="invitation-message"
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontSize": 17,
              },
              Object {
                "fontFamily": "Lato",
              },
              Object {
                "backgroundColor": "#ffffff",
                "color": "#777777",
              },
              Object {
                "fontWeight": "600",
              },
              Object {
                "textAlign": "center",
              },
              Object {
                "backgroundColor": "transparent",
              },
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              Object {
                "marginBottom": 10,
                "marginHorizontal": 5,
                "paddingTop": 10,
              },
            ]
          }
          testID="invitation-message"
        >
          You are now connected to !
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "marginLeft": 15,
              "marginRight": 15,
            },
            undefined,
            Object {
              "marginLeft": 0,
              "marginRight": 0,
            },
            undefined,
          ]
        }
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel="invitation-success-continue"
          accessibilityTraits={undefined}
          accessible={true}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              undefined,
            ]
          }
          testID="invitation-success-continue"
          tvParallaxProperties={undefined}
        >
          <View
            pointerEvents="box-only"
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#9E9E9E",
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "padding": 19,
                },
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                Object {
                  "backgroundColor": "#ffffff",
                },
                undefined,
                Object {
                  "padding": 12,
                },
                undefined,
                undefined,
                undefined,
                Array [
                  null,
                ],
                undefined,
                undefined,
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {},
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  Array [
                    Object {
                      "color": "white",
                      "fontSize": 16,
                    },
                    Object {
                      "color": "#4A8FE3",
                    },
                    Object {
                      "fontSize": 14,
                    },
                    undefined,
                    Object {
                      "fontWeight": "bold",
                    },
                    undefined,
                    undefined,
                    undefined,
                  ],
                ]
              }
            >
              Continue
            </Text>
          </View>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;
